core/src/test/java/org/elasticsearch/cluster/routing/allocation/StartedShardsRoutingTests.java

```diff
@@ -84,4 +87,66 @@ public void testStartedShardsMatching() {
         assertThat(shardRouting.currentNodeId(), equalTo("node2"));
         assertThat(shardRouting.relocatingNodeId(), nullValue());
     }
+
+    public void testRelocatingPrimariesWithInitializingReplicas() {
+        AllocationService allocation = createAllocationService();
+
+        logger.info("--> building initial cluster state");
+        AllocationId primaryId = AllocationId.newRelocation(AllocationId.newInitializing());
+        AllocationId replicaId = AllocationId.newRelocation(AllocationId.newInitializing());
```

Comment:
This doesn't need to be relocating per se? should we use the relocatingReplica boolean?